comp/forwarder/defaultforwarder/worker.go

```diff
@@ -178,6 +205,26 @@ func (w *Worker) ScheduleConnectionReset() {
 	}
 }
 
+// acquireRequestSemaphore attempts to acquire a semaphore, which will block
+// if we are already sending too many requests.
+// This can be bypassed by configuring `forwarder_max_concurrent_requests = 0`.
+func (w *Worker) acquireRequestSemaphore(ctx context.Context) error {
+	if w.maxConcurrentRequests != nil {
+		err := w.maxConcurrentRequests.Acquire(ctx, 1)
+		if err != nil {
+			return err
```

Comment:
Do we not want to wrap these errors to provide additional context (i.e. use `fmt.Errorf`?